controllers/extendeddaemonsetreplicaset/strategy/canary.go

```diff
@@ -154,6 +157,26 @@ func manageCanaryPodFailures(pods []*v1.Pod, params *Parameters, result *Result,
 		cannotStartPodStatus string
 	)
 
+	if canary.AutoFail.MaxRestartsDuration != nil {
+		autoFailMaxRestartsDuration = canary.AutoFail.MaxRestartsDuration
+	}
+
+	if canary.AutoFail.MaxDuration != nil {
+		autoFailMaxDuration = canary.AutoFail.MaxDuration
+
+		// autoFailMaxDuration must be greater than canaryDuration
+		if autoFailMaxDuration.Duration <= canaryDuration.Duration {
+			params.Logger.Info("Canary autoFail.maxDuration must be greater than the canary duration.")
+			if autoPauseEnabled {
+				params.Logger.Info("With AutoPause enabled, canary.autoFail.maxDuration >= 2*canary.duration is recommended.")
+			}
+			autoFailMaxDuration = nil
+		}
```

Comment:
I'm wondering if we should handle this in the CR validation phase, the info log here is useful but if the user doesn't see it will be a silent failure. Or at least we should log it as error and say explicitly that this config will be ignored.

Reply:
yes, i like this! just noticed extendeddaemonset_validate.go , i will move it to there